src/torch_ucc.cpp

```diff
@@ -603,11 +777,16 @@ void CommPG::progress_loop() {
     } catch (...) {
       eptr = std::current_exception();
     }
+    if (work->request_->status == UCC_ERR_TIMED_OUT) {
+      check_communicator_status(work->rank, work->comm_id, work->seq_num_,
+        work->eps);
+    }
     work->finalize(eptr);
```

Comment:
@Sergei-Lebedev - I just realized the CI tests are failing because I suggested to let progress thread aborts immediately in `check_communicator_status` and hence the main thread is not able to catch the exception. I think moving `work->finalize(eptr)` before `check_communicator_status` should fix it. WDYT?

```suggestion
    work->finalize(eptr);
    if (work->request_->status == UCC_ERR_TIMED_OUT) {
      check_communicator_status(work->rank, work->comm_id, work->seq_num_,
        work->eps);
    }
```